projects/kit/components/carousel/carousel-scroller.directive.ts

```diff
@@ -0,0 +1,67 @@
+import {
+    Directive,
+    ElementRef,
+    HostListener,
+    Inject,
+    Input,
+    Optional,
+} from '@angular/core';
+import {INTERSECTION_ROOT} from '@ng-web-apis/intersection-observer';
+import {TUI_IS_MOBILE} from '@taiga-ui/cdk';
+import {shouldCall} from '@tinkoff/ng-event-plugins';
+
+import {TuiCarouselDirective} from './carousel.directive';
+import {TuiCarouselDraggableDirective} from './carousel-draggable.directive';
+
+export function shouldScroll(this: TuiCarouselScrollerDirective): boolean {
+    return !this.isMobile && !!this.draggable?.draggable && this.index === this.current;
+}
+
+// @dynamic
+@Directive({
+    selector: '[tuiCarouselScroller]',
+})
+export class TuiCarouselScrollerDirective {
+    @Input()
+    set scroll(scroll: boolean) {
+        if (scroll) {
+            this.scrollTo(this.directive.inited ? 'smooth' : 'auto');
+        }
+    }
+
+    @Input()
+    index = 0;
+
+    @Input()
+    current = 0;
+
+    constructor(
+        @Optional()
+        @Inject(TuiCarouselDraggableDirective)
+        readonly draggable: TuiCarouselDraggableDirective | null,
+        @Inject(TUI_IS_MOBILE) readonly isMobile: boolean,
```

Comment:
do we also have to do like here?
https://github.com/TinkoffCreditSystems/taiga-ui/pull/1055/files?

Reply:
No, this is not screen dependent but pointer dependent. It is unnecessary on touch devices.